Address thread safety issue in log forwarding

pvollan opened the pull request:
#### 228eebbe8b431b55d009517b892fb427b7d7c51e
<pre>
Address thread safety issue in log forwarding
<a href="https://bugs.webkit.org/show_bug.cgi?id=270366">https://bugs.webkit.org/show_bug.cgi?id=270366</a>
<a href="https://rdar.apple.com/123748756">rdar://123748756</a>

Reviewed by Chris Dumez.

Use dispatch_once to synchronize access to the log worker queue from multiple threads.

* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::logQueue):
(WebKit::registerLogHook):

Canonical link: <a href="https://commits.webkit.org/275573@main">https://commits.webkit.org/275573@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b868a7e1e62c855cc3ad1de3cbe71f65b380242b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/42185 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/21203 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/44579 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/44780 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/38301 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/24401 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/18535 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/34941 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/42759 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/18141 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/36330 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/15878 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/15805 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/46223 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/38382 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/37694 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/41599 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/17001 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/13985 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/40176 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/18620 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/18682 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5680 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/18265 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -802,6 +802,16 @@ static void prewarmLogs()
+    static LazyNeverDestroyed<Ref<WorkQueue>> queue;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, [&] {
+        queue.get() = WorkQueue::create("Log Queue", WorkQueue::QOS::Background);
```
I think this should use:
```
queue.construct(WorkQueue::create("Log Queue", WorkQueue::QOS::Background));
```

pvollan replied:
Ah, nice, will fix.

Thanks for reviewing!

cdumez reviewed (approved):
LGTM with comment.

pvollan committed:
Address thread safety issue in log forwarding
https://bugs.webkit.org/show_bug.cgi?id=270366
rdar://123748756

Reviewed by Chris Dumez.

Use dispatch_once to synchronize access to the log worker queue from multiple threads.

* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::logQueue):
(WebKit::registerLogHook):

Canonical link: https://commits.webkit.org/275573@main
